docs/docs/saving_and_loading_from_catalog.rst

```diff
@@ -56,28 +56,40 @@ To use catalog objects, simply specify their name in the Unitxt object that will
 Modifying Catalog Assets on the Fly
 -----------------------------------
 
-To modify a catalog asset's fields dynamically, use the syntax: `asset.name[key_to_modify=new_value]`. To assign lists, use: `asset.name[key_to_modify=[new_value_0, new_value_1]]`. For instance, to change the metric list of a task:
+To modify a catalog asset's fields dynamically, upon fetching the asset from the catalog, and instantiating it into a python object of type ``unitxt.Artifact``, use the syntax: ``asset.name[key_to_modify=new_value]``. 
+To assign lists, use: ``asset.name[key_to_modify=[new_value_0, new_value_1]]``. 
+To assign dictionaries, use: ``asset.name[key_to_modify={new_key_0=new_value_0,new_key_1=new_value_1}]``.
+Note that the whole new value of the field has to be specified; not just one item of a list, or one key of the dictionary.
+For instance, to change the metric list and the reference specification of a task:
 
 .. code-block:: python
 
     from unitxt.card import TaskCard
 
     card = TaskCard(
         ...
-        task="tasks.my_task[metrics=[metrics.accuracy, metrics.f1[reduction=median]]]"
+        task="tasks.my_task[metrics=[metrics.accuracy, metrics.f1[reduction=median]],reference_fields={output=int}]"
```

Comment:
The dict example is a bit made it, maybe keep the example with just a list?

If we want a more real example, we can use overriding the credential dictionary fields in WMLInferenfeEgnine

Reply:
Agree! removed the dict from the example.  I was not sure what `reduction=median` is, but this is besides the point.